manifest: autoupdate for libenet

muelli opened the pull request.

muelli committed:
manifest: autoupdate for libenet

AsciiWolf commented:
Thanks for your PR! It looks good, but if possible, please remove the x-checker-data part. I prefer to do updates for this Flatpak (that is still based on an outdated Stunt Rally version) by hand.

AsciiWolf commented on org.tuxfamily.StuntRally.yml:
```diff
@@ -149,11 +149,19 @@ modules:
         path: mygui-build.patch
 
-  - name: ENet
+  - name: enet
```
Also, please keep the old module name since other modules are also not in lowercase.

AsciiWolf replied:
(I will change all the module names to lowercase with the SR3 update.)

AsciiWolf commented on org.tuxfamily.StuntRally.yml:
```diff
@@ -149,11 +149,19 @@ modules:
       - type: archive
-        url: http://enet.bespin.org/download/enet-1.3.17.tar.gz
-        sha256: a38f0f194555d558533b8b15c0c478e946310022d0ec7b34334e19e4574dcedc
+        sha256: 28603c895f9ed24a846478180ee72c7376b39b4bb1287b73877e5eae7d96b0dd
```
One more thing: `url` should be above `sha256`.

AsciiWolf replied:
(Or at least this is what it is in all my Flatpaks.)

muelli committed:
flathub: disable f-e-d-c

https://docs.flathub.org/docs/for-app-authors/external-data-checker#disable

In
https://github.com/flathub/org.tuxfamily.StuntRally/pull/26#issuecomment-3541476001
the author expresses the wish to bump dependencies themselves.

AsciiWolf commented on flathub.json:
```diff
@@ -1,3 +1,4 @@
 {
-    "only-arches": ["x86_64"]
+    "only-arches": ["x86_64"],
+    "disable-external-data-checker": true
```
Why disabling the external-data-checker for this branch instead of removing the x-checker-data part from the Flatpak manifest? I don't want to overcomplicate this. :-)

x-checker-data for most modules will be added as part of the 3.0 update (when it is ready).

muelli replied:
Because people come and copy and paste those dependency definitions. And when they come across this, they get the good stuff.

AsciiWolf replied:
Speaking of the 3.x update, if you are able to provide any help, it would be amazing since I was not able to make it work and my free time is sadly quite limited.

Here is its PR (but it is quite outdated and not even converted into yml yet): https://github.com/flathub/org.tuxfamily.StuntRally/pull/17

AsciiWolf replied:
> Because people come and copy and paste those dependency definitions. And when they come across this, they get the good stuff.

Fair point, but please remove it for now. I will archive it as a comment below instead.

It will be added to the 3.x version that will hopefully be ready in upcoming months and will replace the current 2.x Flatpak.

AsciiWolf commented:
Here are the archived enet x-checker-data per my previous comment:
```
        x-checker-data:
          type: anitya
          project-id: 696
          url-template: https://github.com/lsalzman/enet/archive/v$version.tar.gz
```

muelli committed:
manifest: disable x-checker-data for enet as per the maintainer's wish

AsciiWolf commented:
Ping. :-)

AsciiWolf commented:
So... Any chance the x-checker-data could be removed from this PR instead of just disabled?
